pkg/util/log/log.go

```diff
@@ -435,22 +435,22 @@ func logContextWithError(logLevel seelog.LogLevel, bufferFunc func(), scrubAndLo
 
 	isInnerNil := l.inner == nil
 
-	if !isInnerNil && l.shouldLog(logLevel) {
+	if isInnerNil {
+		addLogToBuffer(bufferFunc)
+	} else if l.shouldLog(logLevel) {
 		l.inner.SetContext(context)
 		l.inner.SetAdditionalStackDepth(defaultStackDepth + depth) //nolint:errcheck
 		err := scrubAndLogFunc(message)
 		l.inner.SetContext(nil)
 		l.inner.SetAdditionalStackDepth(defaultStackDepth) //nolint:errcheck
 		defer l.l.Unlock()
 		return err
-	} else if isInnerNil {
-		addLogToBuffer(bufferFunc)
 	}
 
 	l.l.Unlock()
 
 	err := formatErrorc(message, context...)
-	if fallbackStderr {
+	if fallbackStderr && isInnerNil {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Where does this extra condition come from ?
This is a change in behavior, it used to always print if `fallbackStderr` is `true`

Reply:
yes my bad I think my brain stopped working at this moment, I'll rollback this change